validator/sawtooth_validator/journal/genesis.py

```diff
@@ -179,6 +194,34 @@ def start(self, on_done):
         if on_done is not None:
             on_done()
 
+    def _get_block_publisher(self, state_hash):
+        """Returns the block publisher based on the consensus module set by the
+        "sawtooth_config" transaction family.
+
+        Args:
+            state_hash (str): The current state root hash for reading settings.
+
+        Raises:
+            InvalidGenesisStateError: if any errors occur getting the
+                BlockPublisher.
+        """
+        # verify the block with the configured consensus module
+        state_view = self._state_view_factory.create_view(state_hash)
+        config_view = ConfigView(state_view)
+
+        consensus_module_name = config_view.get_setting(
+            'sawtooth.consensus.algorithm', default_value='devmode')
+        try:
+            consensus_module = ConsensusFactory.get_consensus_module(
+                consensus_module_name)
+            return consensus_module.BlockPublisher(
+                BlockCache(self._block_store),
+                state_view=state_view)
+        except UnknownConsensusModuleError:
+            raise InvalidGenesisStateError(
+                'Unable to load consensus module "{}"'.format(
```

Comment:
Should this re-use the msg set on UnknownConsensusModuleError?  "Unable to load consensus module {}: {}".format(consensus_module_name, str(ucme)?